evergreen/archive-evergreen.sh

```diff
@@ -0,0 +1,27 @@
+#!/bin/bash
+set -xeuo pipefail
+
+SRCDIR=$(pwd)/src
+cd $SRCDIR
+# Generate version from the most recent base tag.
+# For example, 1.21.3-4-gdbdcfa34cf is based on the most recent tag "v1.21.3" and
+# has "4" commits on top of that, then it appends the "g" prefix (for git) and
+# an abbreviated object name for the commit. The leading "v" is then omitted.
+# --dirty: If the working tree has local modification "-dirty" is appended to it.
+# --always: Show uniquely abbreviated commit object as fallback when no tag is found.
+VERSION=$(git describe --tags --dirty --always --match 'v[0-9]*' | cut -c 2-)
+cd -
+
+# The archives has to be put in a directory because evergreen s3.put will scan the working directory
+# to find files specified by "local_files_include_filter". However, the build files created by
+# the docker containers cannot be accessed due to permissions, leading to a failure. Changing the
+# working directory to "ARCHIVE_PATH" is a workaround.
+ARCHIVE_PATH=archive
+# Example archive file name: envoy-serverless-1.21.3-4-gdbdcfa34cf.rhel7.amd64.tar.gz
+ARCHIVE_DIR=$ARCHIVE_PATH/envoy-serverless-${VERSION}.rhel7.amd64
```

Comment:
Does it make sense that rhel7 and amd64 are hardcoded here?  Should we name the script file with "rhel7" in the name maybe?

Reply:
Keeping this file generic and using Evergreen expansions would be ideal, but since we only have one build variant here, I hardcoded it. There will be other work to support ARM in the future, do you mind leaving the change to that time?

Reply:
I think we don't need to genericize this script, but it is still prudent to name it with "rhel7" so we remember it's platform-specific.  evergreen/archive-evergreen-rhel7-amd64.sh ?

Reply:
Got it. Done!